sync_telegram/data/mail_sync_project_data.xml

```diff
@@ -16,19 +16,27 @@ def handle_db(records):
         for channel in mail_message.channel_ids:
             if channel.channel_type != 'telegram':
                 continue
-            log("MESSAGE data: channel_id %s - author_id %s - message_text: %s " % (mail_message.res_id, mail_message.author_id.id, mail_message.body), LOG_DEBUG)
-
-            if not html2plaintext(mail_message.body or ""):
-                continue
-
             channel_link = channel.search_links(TG_CHAT_CHANNEL_REL)
             if not channel_link:
                 continue
+
             telegram_user_ref = channel_link.external
 
-            if mail_message.author_id.id in operators_partner_list or mail_message.model != 'mail.channel':
-              telegram_message_html = "%s\n\n<i>%s</i>" % (html_sanitize_telegram(mail_message.body), mail_message.author_id.name.split()[0])
-              telegram.sendMessage(telegram_user_ref, telegram_message_html, parse_mode='HTML')
+            if not mail_message.author_id.search_links(TG_PARTNER_REL):
+                log("MESSAGE from odoo: channel_id %s - author_id %s - attachment_ids: %s - message_text: %s" % (mail_message.res_id, mail_message.author_id.id, mail_message.attachment_ids, mail_message.body), LOG_DEBUG)
+
+                if mail_message.body:
+                    mail_message = "%s\n\n<i>%s</i>" % (mail_message.body, mail_message.author_id.name)
+                    telegram.sendMessage(telegram_user_ref, mail_message, parse_mode='HTML')
+
+                if mail_message.attachment_ids:
```

Comment:
Наблюдаю такую ошибку в логах интеграции:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 330, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 31, in <module>
  File "", line 21, in handle_db
AttributeError: 'str' object has no attribute 'attachment_ids'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 330, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
  File "/mnt/addons/sync-addons/sync/models/sync_trigger_automation.py", line 21, in start
    self.sync_task_id.start(self, args=(records,))
  File "/mnt/addons/sync-addons/sync/models/sync_task.py", line 121, in start
    job, trigger._sync_handler, args, raise_on_error=raise_on_error
  File "/mnt/addons/sync-addons/sync/models/sync_task.py", line 135, in run
    result = self._eval(code, function, args, kwargs, eval_context)
  File "/mnt/addons/sync-addons/sync/models/sync_task.py", line 165, in _eval
    code, eval_context, mode="exec", nocopy=True
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 346, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'AttributeError'>: "'str' object has no attribute 'attachment_ids'" while evaluating
'\ndef handle_db(records):\n    # records are instances of mail.message\n    for mail_message in records:\n        for channel in mail_message.channel_ids:\n            if channel.channel_type != \'telegram\':\n                continue\n            channel_link = channel.search_links(TG_CHAT_CHANNEL_REL)\n            if not channel_link:\n                continue\n\n            telegram_user_ref = channel_link.external\n\n            if not mail_message.author_id.search_links(TG_PARTNER_REL):\n                log("MESSAGE from odoo: channel_id %s - author_id %s - attachment_ids: %s - message_text: %s" % (mail_message.res_id, mail_message.author_id.id, mail_message.attachment_ids, mail_message.body), LOG_DEBUG)\n\n                if mail_message.body:\n                    mail_message = "%s\\n\\n<i>%s</i>" % (mail_message.body, mail_message.author_id.name)\n                    telegram.sendMessage(telegram_user_ref, mail_message, parse_mode=\'HTML\')\n\n                if mail_message.attachment_ids:\n                    for attach in mail_message.attachment_ids:\n                        log("ATACHMENT from odoo attach.id %s - name %s" % (attach.name, attach.id), LOG_INFO)\n\n                        if attach.mimetype and "/" in attach.mimetype and attach.mimetype.split("/")[0] == "image":\n                            telegram.sendPhoto(telegram_user_ref, attach.datas)\n                        else:\n                            telegram.sendDocument(telegram_user_ref, attach.name, attach.datas)\n\n        \nEXECUTION_RESULT_ = handle_db(*EXECUTION_ARGS_, **EXECUTION_KWARGS_)\n        '

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 330, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 65, in <module>
  File "", line 49, in handle_webhook
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_channel.py", line 416, in message_post
    message = super(Channel, self.with_context(mail_create_nosubscribe=True)).message_post(message_type=message_type, moderation_status=moderation_status, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 1888, in message_post
    self._notify_thread(new_message, values, **notif_kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_channel.py", line 542, in _notify_thread
    super(Channel, self)._notify_thread(message, msg_vals=msg_vals, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/sms/models/mail_thread.py", line 236, in _notify_thread
    recipients_data = super(MailThread, self)._notify_thread(message, msg_vals=msg_vals, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 2135, in _notify_thread
    self._notify_record_by_inbox(message, rdata, msg_vals=msg_vals, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 2153, in _notify_record_by_inbox
    message.write({'channel_ids': [(6, 0, channel_ids)]})
  File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 348, in write
    action._process(records, domain_post=domain_post)
  File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 273, in _process
    raise e
  File "/usr/lib/python3/dist-packages/odoo/addons/base_automation/models/base_automation.py", line 270, in _process
    self.action_server_id.sudo().with_context(**ctx).run()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_actions.py", line 629, in run
    res = runner(run_self, eval_context=eval_context)
  File "/usr/lib/python3/dist-packages/odoo/addons/website/models/ir_actions.py", line 61, in _run_action_code_multi
    res = super(ServerAction, self)._run_action_code_multi(eval_context)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_actions.py", line 498, in _run_action_code_multi
    safe_eval(self.code.strip(), eval_context, mode="exec", nocopy=True)  # nocopy allows to return 'action'
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 346, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'ValueError'>: "<class 'AttributeError'>: "'str' object has no attribute 'attachment_ids'" while evaluating
'\ndef handle_db(records):\n    # records are instances of mail.message\n    for mail_message in records:\n        for channel in mail_message.channel_ids:\n            if channel.channel_type != \'telegram\':\n                continue\n            channel_link = channel.search_links(TG_CHAT_CHANNEL_REL)\n            if not channel_link:\n                continue\n\n            telegram_user_ref = channel_link.external\n\n            if not mail_message.author_id.search_links(TG_PARTNER_REL):\n                log("MESSAGE from odoo: channel_id %s - author_id %s - attachment_ids: %s - message_text: %s" % (mail_message.res_id, mail_message.author_id.id, mail_message.attachment_ids, mail_message.body), LOG_DEBUG)\n\n                if mail_message.body:\n                    mail_message = "%s\\n\\n<i>%s</i>" % (mail_message.body, mail_message.author_id.name)\n                    telegram.sendMessage(telegram_user_ref, mail_message, parse_mode=\'HTML\')\n\n                if mail_message.attachment_ids:\n                    for attach in mail_message.attachment_ids:\n                        log("ATACHMENT from odoo attach.id %s - name %s" % (attach.name, attach.id), LOG_INFO)\n\n                        if attach.mimetype and "/" in attach.mimetype and attach.mimetype.split("/")[0] == "image":\n                            telegram.sendPhoto(telegram_user_ref, attach.datas)\n                        else:\n                            telegram.sendDocument(telegram_user_ref, attach.name, attach.datas)\n\n        \nEXECUTION_RESULT_ = handle_db(*EXECUTION_ARGS_, **EXECUTION_KWARGS_)\n        '" while evaluating
'env["sync.trigger.automation"].browse(4).sudo().start(records)'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mnt/addons/sync-addons/sync/models/sync_task.py", line 135, in run
    result = self._eval(code, function, args, kwargs, eval_context)
  File "/mnt/addons/sync-addons/sync/models/sync_task.py", line 165, in _eval
    code, eval_context, mode="exec", nocopy=True
  File "/usr/lib/python3/dist-packages/odoo/tools/safe_eval.py", line 346, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'ValueError'>: "<class 'ValueError'>: "<class 'AttributeError'>: "'str' object has no attribute 'attachment_ids'" while evaluating
'\ndef handle_db(records):\n    # records are instances of mail.message\n    for mail_message in records:\n        for channel in mail_message.channel_ids:\n            if channel.channel_type != \'telegram\':\n                continue\n            channel_link = channel.search_links(TG_CHAT_CHANNEL_REL)\n            if not channel_link:\n                continue\n\n            telegram_user_ref = channel_link.external\n\n            if not mail_message.author_id.search_links(TG_PARTNER_REL):\n                log("MESSAGE from odoo: channel_id %s - author_id %s - attachment_ids: %s - message_text: %s" % (mail_message.res_id, mail_message.author_id.id, mail_message.attachment_ids, mail_message.body), LOG_DEBUG)\n\n                if mail_message.body:\n                    mail_message = "%s\\n\\n<i>%s</i>" % (mail_message.body, mail_message.author_id.name)\n                    telegram.sendMessage(telegram_user_ref, mail_message, parse_mode=\'HTML\')\n\n                if mail_message.attachment_ids:\n                    for attach in mail_message.attachment_ids:\n                        log("ATACHMENT from odoo attach.id %s - name %s" % (attach.name, attach.id), LOG_INFO)\n\n                        if attach.mimetype and "/" in attach.mimetype and attach.mimetype.split("/")[0] == "image":\n                            telegram.sendPhoto(telegram_user_ref, attach.datas)\n                        else:\n                            telegram.sendDocument(telegram_user_ref, attach.name, attach.datas)\n\n        \nEXECUTION_RESULT_ = handle_db(*EXECUTION_ARGS_, **EXECUTION_KWARGS_)\n        '" while evaluating
'env["sync.trigger.automation"].browse(4).sudo().start(records)'" while evaluating
'\ndef handle_webhook(httprequest):\n  data = json.loads(httprequest.data.decode("utf-8"))\n  log("Raw data: %s" % data, LOG_DEBUG)\n\n  update = telegram.parse_data(data)\n  message = update.message or update.edited_message\n  odoo_message_text = message.text or message.caption or ""\n  name = params.PARTNER_NAME_TEMPLATE % user2name(message.from_user)\n  record_name = params.CHAT_RECORD_NAME_TEMPLATE % user2name(message.from_user)\n  telegram_user_ref =  message.from_user.id\n  odoobot = env.user.browse(1).partner_id.id\n  main_operator = env.user.browse(operators_user_list[0])\n\n  partner_link = get_link(TG_PARTNER_REL, telegram_user_ref)\n\n  if not partner_link:\n    partner_image =  telegram.getUserPhoto(telegram_user_ref)\n    partner = env["res.partner"].with_user(main_operator).create({"name": name, "image_1920": partner_image})\n    partner.set_link(TG_PARTNER_REL, telegram_user_ref, sync_date=None, allow_many2many=False)\n    partner_link = get_link(TG_PARTNER_REL, telegram_user_ref)\n\n  if params.WEBHOOK_CHAT != "no":\n    if message.text == "/start":\n      telegram.sendMessage(telegram_user_ref, texts.TELEGRAM_WELCOME_MESSAGE)\n    elif bool(update.edited_message):\n      odoo_message_text = "Contact has edited the message:"\n\n    channel_link = get_link(TG_CHAT_CHANNEL_REL, telegram_user_ref)\n    if channel_link:\n      channel = channel_link.odoo\n    else:\n      partners = [partner_link.odoo.id] + operators_partner_list\n      channel = telegram.create_mail_сhannel(partners)\n      channel.set_link(TG_CHAT_CHANNEL_REL, telegram_user_ref)\n\n    model_rel = TG_CHAT_MODEL_REL + params.CHAT_MODEL\n    if params.CHAT_MODEL != "res.partner":\n      record_model_link = get_link(model_rel, telegram_user_ref)\n\n      if not record_model_link:\n        record_model = env[params.CHAT_MODEL].with_user(main_operator).create({"name": record_name, "partner_id": partner_link.odoo.id})\n        record_model.set_link(model_rel, telegram_user_ref, sync_date=None, allow_many2many=False)\n\n        record_model.message_post(body="""<b><a href="/web#action=%s&active_id=mail.channel_%s">Chat with partner is created. </a></b>""" %\n        (env.ref("mail.action_discuss").id, channel.id), author_id=odoobot, message_type="comment", subtype_xmlid="mail.mt_comment")\n\n        channel.message_post(body="""<b><a href="/web#id=%s&model=%s">Record is created. </a></b>""" %\n        (record_model.id, record_model._name), author_id=odoobot, message_type="comment", subtype_xmlid="mail.mt_comment")\n\n    file_bin_data = []\n    telegram_file = update.message.document or update.message.voice or update.message.sticker\n    if update.message.photo:\n        #Gets the largest size instead of the preview image\n        file_bin_data = telegram.getFileBinary(telegram_user_ref, update.message.photo[-1])\n    elif telegram_file:\n        file_bin_data = telegram.getFileBinary(telegram_user_ref, telegram_file)\n\n    channel.message_post(body=odoo_message_text, attachments=[file_bin_data], author_id=partner_link.odoo.id,\n    message_type="comment", subtype_xmlid="mail.mt_comment")\n\n\n\n        \nEXECUTION_RESULT_ = handle_webhook(*EXECUTION_ARGS_, **EXECUTION_KWARGS_)\n        '
```